trl/experimental/gold/gold_trainer.py

```diff
@@ -1916,7 +1919,9 @@ def training_step(
                 )
                 new_input_ids, new_attention_mask, new_labels, prompt_texts, completion_texts = result
             else:
-                with unwrap_model_for_generation(model, self.accelerator) as unwrapped_model:
+                with unwrap_model_for_generation(
+                    model, self.accelerator, generation_kwargs=self.generation_kwargs
+                ) as unwrapped_model:
                     result = self.generate_on_policy_outputs(
                         unwrapped_model, inputs, self.generation_config, self.processing_class.pad_token_id
```

Comment:
When reading the code, it's not clear why we need both a context with `generation_kwargs=self.generation_kwargs` + passing the generation config. Can you possibly add a comment, and maybe a link to the GitHub issue(s)?